components/contract-components/hooks.ts

```diff
@@ -91,7 +91,12 @@ export function usePublishMutation() {
       sdk && "getPublisher" in sdk,
       "sdk is not ready or does not support publishing",
     );
-    return await (await sdk.getPublisher()).publishBatch(uris);
+    // FIXME - can't call batch here because of a bug with multicall + gasless setup. Looping through the uris and calling publish one by one for now.
+    const receipts = [];
+    for (const uri of uris) {
+      receipts.push(await sdk.getPublisher().publish(uri));
+    }
+    return receipts;
```

Comment:
would prefer `Promise.all()` here.

I believe doing it in a for loop will mean it's sequential instead of simultaneous?

I guess you already have to sign each individually so it shouldn't matter? (But still would prefer Promise.all()

Reply:
meant to change this before merging - will do a new pr for it